src/track/track.cpp

```diff
@@ -70,7 +71,11 @@ Track::Track(
           m_record(trackId),
           m_bDirty(false),
           m_bMarkedForMetadataExport(false),
-          m_analyzerProgress(-1) {
+          m_analyzerProgress(-1),          
```

Comment:
I don't think that it is appropriate to store transient state of an application component directly inside the Track object. For example m_analyzerProgress does not belong in here and will be removed soon (#1624).

You should keep track of the tracks that are currently played in your component and manage this information there. Connect to the PlayerManager or individual decks for getting notified about which tracks you need to observe.

Reply:
I don't get what you mean exactly? All info I've put in track is the number of ms it has been played along with some timers to update this number and whether if the track is scrobbable or not. Doesn't this information belong to the track? Should I keep a list of the tracks being currently played and update their played time in the player manager?

Reply:
The Track object is already huge and becomes unmanageable if every aspect is thrown in there. It should contain only persistent state that needs to be synchronized with the database and the corresponding file, but not arbitrary transient application state.

Please try to keep the state that is needed for live metadata broadcasting private and local within the corresponding component or module. Use signals/slots for synchronization instead of distributing implementation details into public, shared global state.

Think in terms of modular microservices, even in the context of a desktop application. Consume signals/notifications from other components as input and produce output by emitting signals. All transient state should be isolated within the corresponding module.

Reply:
Okey, I get it now. I will move my logic away from Track.